addons/mrp/models/mrp_production.py

```diff
@@ -674,7 +674,9 @@ def action_confirm(self):
                 })
             production._generate_finished_moves()
             production.move_raw_ids._adjust_procure_method()
-            (production.move_raw_ids | production.move_finished_ids)._action_confirm()
+            (production.move_raw_ids | production.move_finished_ids).with_context(
+                skip_phantom_moves=production.move_finished_ids.ids
+            )._action_confirm()
```

Comment:
We should still explode byproduct. I would do the whole patch in action_explode with a check.
``` python3
if move.production_id and move.production_id.product_id == move.product_id:
    moves_to_return |= move
    continue
```